pkg/asset/machines/openstack/openstackmachines.go

```diff
@@ -186,23 +186,30 @@ func generateMachineSpec(clusterID string, platform *openstack.Platform, mpool *
 		CloudName: CloudName,
 		Flavor:    mpool.FlavorName,
 		IdentityRef: &capo.OpenStackIdentityReference{
-			Kind: "Secret",
 			Name: clusterID + "-cloud-config",
 		},
 		// FIXME(stephenfin): We probably want a FIP for bootstrap?
-		// TODO: This is an image name. Migrate to a filter with Name when API v1alpha8 is released.
-		Image:          osImage,
+		Image: capo.ImageFilter{
+			Name: osImage,
+		},
 		Ports:          append([]capo.PortOpts{port}, additionalPorts...),
 		SecurityGroups: securityGroups,
 		// FIXME: Uncomment when the server group rework merged
 		// https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1779
 		//ServerGroup: *capo.ServerGroupFilter{
```

Comment:
FIltering serverGroup by Name was added into CAPO, so we can start enforcing it.